### PR TITLE
Responsive charts with new Resizer component

### LINKED ISSUE #442: Make charts responsive
width and height do not work as you would expect with SVGs. SVGs as responsive components care more about the `preserveAspectRatio` and `viewbox` attributes. The calculations currently being used to set width and height need to be used to set viewbox instead. 
Width and height should be passthrough props instead.

### PR DESCRIPTION
**Pull request by jondlm**
Fixes #442

## PR Checklist

- Manually tested across supported browsers
  - [x] Chrome
  - [x] Firefox
  - [x] Safari
  - [x] IE11 (Win 7)
  - [x] Edge (Win 10)
- [x] Unit tests written (`common` at minimum)
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [ ] One core team UX approval

### CODE REVIEW

File: src/components/BarChart/BarChart.jsx [hunk 1]
```diff
@@ -280,13 +280,15 @@ const BarChart = createClass({
 			...marginOriginal,
 		};
 
+		const svgClasses = cx(className, '&');
+
 		// TODO: Consider displaying something specific when there is no data,
 		// perhaps a loading indicator.
-		if (_.isEmpty(data)) {
+		if (_.isEmpty(data) || width < 1 || height < 1) {
```
Comment (jondlm) [inline src/components/BarChart/BarChart.jsx]: I added these just to guard against the case where charts might be rendered with 0 or negative widths/heights.

File: webpack.config.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ var gulpConfig = require('./gulp/config');
 var CopyWebpackPlugin = require('copy-webpack-plugin');
 
 module.exports = {
-	devtool: 'source-map',
+	devtool: 'eval',
```
Comment (jondlm) [inline webpack.config.js]: Snuck in this, makes webpack watching way faster but might make source maps a little worse.

File: src/components/Resizer/Resizer.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { mount } from 'enzyme';
+
+import Resizer, { __RewireAPI__ as rewire } from './Resizer';
+
+describe('Resizer', () => {
+	common(Resizer, {
+		exemptFunctionProps: ['children'],
+		getDefaultProps: () => {
+			return {
+				children: () => (<div></div>),
```
Comment (ogupte) [inline src/components/Resizer/Resizer.spec.jsx]: `(<div />)`

File: webpack.config.js (hunk 2 above)
Comment (alisonailea) [inline webpack.config.js]: What about eval-source-map?

File: src/components/BarChart/examples/4.grouped-with-legend.jsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import _ from 'lodash';
```
Comment (jondlm) [inline src/components/BarChart/examples/4.grouped-with-legend.jsx]: I threw something into this PR that I'd been meaning to do for a while: removing the examples with embedded legends and replacing them with the examples showing how to use the `Legend` component directly.

File: src/components/Resizer/Resizer.jsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		const {
+			width,
+			height,
+		} = this.state;
+
+		return (
+			<div
+				className={cx('&', className)}
+				ref={(ref) => this._element = ref}
+				{...passThroughs}
```
Comment (sodiumjoe) [inline src/components/Resizer/Resizer.jsx]: `omitProps`? doesn't really do anything here, but might be good to add it just to guard against future props being added

File: src/components/Resizer/Resizer.jsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+			>
+				{this.props.children(width, height)}
+			</div>
+		);
+	},
+});
+
+export default Resizer;
+
```
Comment (sodiumjoe) [inline src/components/Resizer/Resizer.jsx]: extra new line, sorry pet peeve

File: src/components/Resizer/Resizer.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+	describe('props', () => {
+		it('children should callback with width and height', () => {
+			const erd = {
+				listenTo: (_element, handleResize) => {
+					handleResize({
+						offsetWidth: 50,
+						offsetHeight: 100,
+					});
+				},
+				removeListener: sinon.spy(),
```
Comment (sodiumjoe) [inline src/components/Resizer/Resizer.spec.jsx]: were you planning to add a test that uses this?

Comment (jondlm) [inline src/components/Resizer/Resizer.spec.jsx]: Oh yeah! I forgot to write the unmount test.

**Pull request merged**